cmd/koord-manager/options/options.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package options
+
+import (
+	"flag"
+	"fmt"
+	"strings"
+
+	"github.com/spf13/pflag"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+type Options struct {
+	ControllerAddFuncs map[string]func(manager.Manager) error
+	Controllers        []string
+}
+
+func NewOptions() *Options {
+	return &Options{
+		ControllerAddFuncs: controllerAddFuncs,
+		Controllers:        sets.StringKeySet(controllerAddFuncs).List(),
+	}
+}
+
+func (o *Options) InitFlags(fs *flag.FlagSet) {
+	pflag.StringSliceVar(&o.Controllers, "controllers", o.Controllers, fmt.Sprintf("A list of controllers to enable. '*' enables all on-by-default controllers,"+
```

Comment:
add some explanation of full examples: 
case1: -controllers=*,-noderesource, means enables all controllers except noderesource
case2: -controllers=noderesource, means only enables noderesource

Reply:
done